### PR TITLE
Abstract tokio runtime shut down check

### PR DESCRIPTION
**Pull request by hzxuzhonghu**

### CODE REVIEW

File: src/proxy/util.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::io::{Error, ErrorKind};
+
+pub fn is_runtime_shutdown(e: &Error) -> bool {
+    match e.kind() {
+        ErrorKind::Other => {
```
Comment (hzxuzhonghu) [inline src/proxy/util.rs:19]: I am not sure which is the idiotic way for rust

`match vs if`

Comment (PiotrSikora) [reply src/proxy/util.rs:19]: Since we care about both conditions, I'd go with `if ... && ...` instead of this `match` and `==` mix.

File: src/workload.rs [hunk 2]
```diff
@@ -309,13 +309,12 @@ impl WorkloadInformation {
     pub async fn fetch_workload(&self, addr: &IpAddr) -> Option<Workload> {
```
Comment (PiotrSikora) [inline src/workload.rs:309]: All those changes in `workload.rs` seem unrelated.

Comment (hzxuzhonghu) [reply src/workload.rs:309]: These are meant to be commited to https://github.com/istio/ztunnel/pull/262

File: src/proxy/util.rs (hunk 1 above)
Comment (hzxuzhonghu) [reply src/proxy/util.rs:19]: ok

**Review by howardjohn — APPROVED** [review]
